lib/polyn/application.rb

```diff
@@ -86,8 +88,18 @@ def on_message((msg, *args))
       case msg
       when :publish
         publish(*args)
+      when :terminated
+        begin
+          logger.fatal("there was a problem with initializing the Polyn application, shutting down" \
+                         " Check logs for details.")
+          transit.shutdown
+          reactor_manager&.shutdown
+        ensure
+          puts "Polyn forced the application to exit due to an error."
+          Process.kill("INT", Process.pid)
+        end
```

Comment:
This should be moved to a private method. Its an implementation detail. I am not a huge fan of case statements for this reason.